### PR TITLE
5708 vet360 linking

### PR DESCRIPTION
**Pull request by kpethtel**
## Summary
This ticket fixes two issues. 
1) it moves the vet360 linking out of the authenticate method. The authenticate method is called on every single request, which meant that when a vet is using the app and they don't have a vet360_id, sometimes multiple linking jobs would be spun up per session. The redis locking mechanism did not appear to reliably prevent this.
2) it avoids the poll with backoff functionality. That functionality (which is still used for synchronous contact update requests) uses the sleep method. Most of the time that's not a problem because the request usually works the first try, but when the upstream is down, a single job will occupy a sidekiq thread for a full 55 seconds. 

Because of these two issues, when the upstream is down, we could see multiple jobs stack up per user and occupy sidekiq threads for nearly a minute each, which puts a lot of unnecessary strain on sidekiq. 

Another notable side-effect of this change is that now the linking will happen on both IAM and SIS requests. Previously, these requests were only happening on IAM sessions because no one knew what the linking did so we didn't bother reproducing it for SIS. The fact that this will potentially happen more is a good thing because some mobile app features don't work without a VA Profile account, which the linking job attempts to create for users who don't have them. So this should result in a better mobile app experience for some users. 

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/5708

## Testing done

Specs

## What areas of the site does it impact?
- No major impact. The linking job may occur more frequently but should also tie up threads less often.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

Any and all.

### CODE REVIEW

File: modules/mobile/spec/request/v1/user_request_spec.rb [hunk 1]
```diff
@@ -299,7 +299,7 @@
           VCR.use_cassette('mobile/payment_information/payment_information') do
             VCR.use_cassette('mobile/user/get_facilities') do
               VCR.use_cassette('mobile/va_profile/demographics/demographics') do
-                get '/mobile/v0/user', headers: iam_headers
```
Comment (kpethtel) [inline modules/mobile/spec/request/v1/user_request_spec.rb:302]: These are not related to the PR. They were mistakes that were made when creating the v1 of this spec.

File: modules/mobile/app/workers/mobile/v0/vet360_linking_job.rb [hunk 2]
```diff
@@ -15,16 +28,13 @@ def perform(uuid)
         user = IAMUser.find(uuid) || User.find(uuid)
         raise MissingUserError, uuid unless user
 
-        result = Mobile::V0::Profile::SyncUpdateService.new(user).await_vet360_account_link
-        Rails.logger.info('Mobile Vet360 account linking succeeded for user with uuid',
-                          { user_uuid: uuid, transaction_id: result.transaction_id })
+        result = VAProfile::Person::Service.new(user).init_vet360_id
+        Rails.logger.info('Mobile Vet360 account linking request succeeded for user with uuid',
+                          { user_uuid: uuid, transaction_id: result.transaction.id })
```
Comment (kpethtel) [inline modules/mobile/app/workers/mobile/v0/vet360_linking_job.rb:33]: I don't think there's any value in this transaction id. One thing that may not be very clear from the diff is that this transaction is different from the transaction that was previously being returned. Neither of them truly valuable unless we want to digging deeply into logs with the VAProfile team.

**Review by aherzberg — APPROVED** [review]
LGTM.

**Review by ryan-mcneil — APPROVED** [review]
👌
